src/test/java/io/appium/java_client/android/AndroidConnectionTest.java

```diff
@@ -26,23 +28,32 @@
 public class AndroidConnectionTest extends BaseAndroidTest {
 
     @Test public void test1() {
-        driver.setConnection(Connection.WIFI);
-        assertEquals(Connection.WIFI,
-            driver.getConnection());
+        driver.setConnection(new ConnectionStateBuilder()
+                .withWiFiEnabled()
+                .build());
+        assertTrue(driver.getConnection().isWiFiEnabled());
     }
 
     @Test public void test2() {
-        driver.setConnection(Connection.NONE);
-        assertEquals(Connection.NONE,
-            driver.getConnection());
-        driver.setConnection(Connection.AIRPLANE);
-        assertEquals(Connection.AIRPLANE,
-            driver.getConnection());
+        driver.setConnection(new ConnectionStateBuilder()
+                .withAirplaneModeDisabled()
+                .build());
+        ConnectionState state = driver.getConnection();
+        assertTrue(!state.isAirplaneModeEnabled() && !state.isWiFiEnabled() && !state.isDataEnabled());
```

Comment:
check for wifi and data is not necessary i believe as disabling airplane mode will only disable airplane mode incase if it was already enabled and leave's the rest of connection as is. Isn't it?

Enabling airplane mode will disable both data and wifi but viseversa is not true. Correct me if i'm wrong.

Reply:
Here also we might need to remove the additional checks.

Reply:
These I'd prefer to keep